test/WebJobs.Script.Tests/Workers/Rpc/RpcWorkerProcessTests.cs

```diff
@@ -42,6 +51,78 @@ public RpcWorkerProcessTests()
                 serviceProviderMock.Object);
         }
 
+        [Fact]
+        public void Constructor_RegistersHostStartedEvent()
+        {
+            Func<IObserver<ScriptEvent>, bool> validate = p =>
+            {
+                // validate the internal reactive ISink<HostStartEvent> implementation that
+                // results from the Subscribe extension method
+                Type eventType = p.GetType().GetInterfaces()[0].GetGenericArguments()[0];
+                return eventType == typeof(HostStartEvent);
+            };
+
+            _eventManager.Verify(_ => _.Subscribe(It.Is<IObserver<ScriptEvent>>(p => validate(p))), Times.Once());
+        }
+
+        [Fact]
+        public void OnHostStart_RegistersProcessWithMonitor()
+        {
+            Process process = Process.GetCurrentProcess();
+            _rpcWorkerProcess.Process = process;
+
+            _hostProcessMonitorMock.Setup(p => p.RegisterChildProcess(process));
+
+            HostStartEvent evt = new HostStartEvent();
+            _rpcWorkerProcess.OnHostStart(evt);
+            _hostProcessMonitorMock.Verify(p => p.RegisterChildProcess(process), Times.Once);
+        }
+
+        [Fact]
+        public void RegisterWithProcessMonitor_Succeeds()
+        {
+            Process process = Process.GetCurrentProcess();
+            _rpcWorkerProcess.Process = process;
+
+            _hostProcessMonitorMock.Setup(p => p.RegisterChildProcess(process));
+
+            _rpcWorkerProcess.RegisterWithProcessMonitor();
+            _hostProcessMonitorMock.Verify(p => p.RegisterChildProcess(process), Times.Once);
+
+            // registration is skipped if attempted again for the same monitor
+            _rpcWorkerProcess.RegisterWithProcessMonitor();
+            _hostProcessMonitorMock.Verify(p => p.RegisterChildProcess(process), Times.Once);
+
+            // if the monitor changes (e.g. due to a host restart)
+            // registration is performed
+            _hostProcessMonitorMock = new Mock<IHostProcessMonitor>(MockBehavior.Strict);
+            _hostProcessMonitorMock.Setup(p => p.RegisterChildProcess(process));
```

Comment:
this is good test coverage! I recommend updating this test to publish `HostStartEvent` and verifying `RegisterWithProcessMonitor` .

Reply:
The event manager used by this test suite is a mock, so can't be used to publish. However I have test validating that the constructor registers the event handler and that when that handler (OnHostStart) is called it does the right thing. I'm intentionally not testing that IScriptEventManager delivers an event when it's published - that's outside of the scope of these unit tests (i.e. I'm not testing IScriptEventManager).